src/Parser/Druid/Feral/Modules/FeralCore/Snapshot.js

```diff
@@ -0,0 +1,160 @@
+import SPELLS from 'common/SPELLS';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import { encodeTargetString } from 'Parser/Core/Modules/EnemyInstances';
+
+const debug = false;
+
+/*
+Feral has a snapshotting mechanic which means the effect of some buffs are maintained over the duration of
+some DoTs even after the buff has worn off.
+Players should follow a number of rules with regards when they refresh a DoT and when they do not, depending
+on what buffs the DoT has snapshot and what buffs are currently active.
+
+RAKE snapshots:
+  Prowl, Tiger's Fury, Bloodtalons
+RIP snapshots:
+  Tiger's Fury, Bloodtalons
+MOONFIRE_FERAL snapshots:
+  Tiger's Fury
+THRASH_FERAL snapshots: (but isn't used in single target situations)
+  Tiger's Fury, Moment of Clarity
+
+The Snapshot class is 'abstract', and shouldn't be directly instantiated. Instead classes should extend
+it and test how well the combatant is making use of the snapshot mechanic.
+*/
+
+const TIGERS_FURY_MULTIPLIER = 1.15;
+const PROWL_MULTIPLIER = 2.00; // also applied by Incarnation: King of the Jungle, and Shadowmeld
+const BLOODTALONS_MULTIPLIER = 1.20;
+
+const JAGGED_WOUNDS_MODIFIER = 0.80;  // "[...]deal the same damage as normal but in 20% less time."
+const PANDEMIC_FRACTION = 0.3;
+
+// leeway in ms after loss of bloodtalons/prowl buff to count a cast as being buffed. Keep as low as possible to avoid false positives from buffs fading due to causes other than being used to buff a DoT.
+const BUFF_WINDOW_TIME = 60;
+// leeway in ms between a cast event and debuff apply/refresh for them to be associated
+const CAST_WINDOW_TIME = 100;
+
+class Snapshot extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  // extending class should fill these in:
+  spellCastId;
+  debuffId;
+  isProwlAffected;
+  isTigersFuryAffected;
+  isBloodtalonsAffected;
+  durationOfFresh;
+
+  stateByTarget = [];
+  lastDoTCastEvent;
+
+  on_byPlayer_cast(event) {
+    if (this.spellCastId !== event.ability.guid) {
+      return;
+    }
+    this.lastDoTCastEvent = event;
+    
+    // attempt to associate with (very) recent debuff apply/refresh
+    // this cast could apply DoTs to multiple targets, so check them all
+    for (const state of this.stateByTarget) {
+      if (!state.castEvent &&
+          event.timestamp - state.startTime < CAST_WINDOW_TIME) {
+        event.castEvent = event;
+        this.checkRuleIfReady(event, state.prev, state);
```

Comment:
method only takes 1 parameter, I assume you forgot to remove the last two?

Reply:
Oooh, yikes. I actually forgot to remove the *first* two. It should be:
`this.checkRuleIfReady(state);`
I've pushed a fix into my branch. Will I need to make a separate pull request as this has been merged now?

Reply:
Yeah you'll have to make a new PR for the fix, I'll merge in as soon as Travis is done :)

Reply:
Thank you again. Should be ready now #1752